src/main/java/net/openhft/chronicle/network/RemoteConnector.java

```diff
@@ -168,6 +168,13 @@ public boolean action() throws InvalidEventHandlerException {
                     // this can happen if the acceptor is in the process of shutting down
                     return false;
 
+                if (nc.isClosed()) {
+                    // may be already closed by socketReconnector in HostConnector
+                    closeSocket(sc);
+
+                    throw InvalidEventHandlerException.reusable();
+                }
+
                 nc.socketChannel(sc);
```

Comment:
Next calls still can throw `ClosedIllegalStateException` if `nc` can be closed concurrently.
It looks safer to do this in `ClosedIllegalStateException`'s catch block.

Reply:
Can you check now? The impact is greater if the check is moved to catch block.